cosmos/scheduler/cmd/scheduler.go

```diff
@@ -32,33 +34,50 @@ type Scheduler struct {
 	IngestionSchedule string
 }
 
+// BucketTransaction is used as the decoded element returned by MongoDB
+type BucketTransaction struct {
+	LocationType  string
+	OperationType string `bson:"operationType"`
+	FullDocument  struct {
+		Value struct {
+			Name               string `bson:"name"`
+			OwnerDisplayName   string `bson:"ownerDisplayName"`
+			Deleted            bool   `bson:"deleted"`
+			LocationConstraint string `bson:"locationConstraint"`
+		} `bson:"value"`
+	} `bson:"fullDocument"`
+}
+
 // Run starts the Cosmos Scheduler
 func (s *Scheduler) Run() {
 	log.Println("starting cosmos scheduler")
-	quit := make(chan bool)
-	overlayUpdates, err := s.watchOverlayUpdates()
-	if err != nil {
-		log.Println(err)
-		return
-	}
 	go s.healthCheckServer()
-	err = s.configureIngestionSecret(true)
+
+	// safely create or update secret on startup
+	err := s.configureIngestionSecret(true)
 	if err != nil{
-		log.Fatal("error configuring secret:", err)
+		log.Fatal("run failed configuring secret: ", err)
 	}
+
+	overlayUpdates := s.watchOverlayUpdates()
+
+	quit := make(chan bool)
 	for {
-		go s.watchBucketUpdates(quit)
-		<-overlayUpdates
+		go s.watchBucketUpdates("nfs", quit)
+		// blocks until an overlay update is recieved on the overlayUpdates channel
+		<- overlayUpdates
 		log.Println("received an overlay update")
 		quit <- true
```

Comment:
Why quit here? It seems that this is doing "watch, receive, quit" in a loop, which has the potential to miss events because of the asynchronicity involved.

Reply:
Agreed, I think I can more easily rework this now with your previous comments.